itou/utils/urls.py

```diff
@@ -57,3 +60,17 @@ def to_python(self, value):
 
     def to_url(self, value):
         return f"{value}"
+
+
+# Metabase private / signed URLs
+# See:
+# * https://www.metabase.com/docs/latest/enterprise-guide/full-app-embedding.html
+# * https://github.com/jpadilla/pyjwt
+# * https://github.com/metabase/embedding-reference-apps/blob/master/django/embedded_analytics/user_stats/views.py
+def _get_token(payload):
+    return jwt.encode(payload, settings.METABASE_SECRET_KEY, algorithm="HS256").decode("utf8")
+
+
+def metabase_embedded_url(dashboard_id):
+    payload = {"resource": {"dashboard": dashboard_id}, "params": {}, "exp": round(time.time()) + (60 * 10)}
+    return settings.METABASE_SITE_URL + "/embed/dashboard/" + _get_token(payload)
```

Comment:
Je voyais ça plutôt dans`itou.utils.apis.metabase`.

Reply:
Why not